src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -448,78 +448,60 @@ protected override void Generate(AlterColumnOperation operation, IModel model, M
             EndStatement(builder);
         }
 
-        protected override void Generate(RenameIndexOperation operation, [CanBeNull] IModel model, MigrationCommandListBuilder builder)
+        protected override void Generate(RenameIndexOperation operation, IModel model, MigrationCommandListBuilder builder)
         {
             Check.NotNull(operation, nameof(operation));
             Check.NotNull(builder, nameof(builder));
 
-            var qualifiedName = new StringBuilder();
-            if (operation.Schema != null)
+            if (operation.NewName != null &&
+                operation.NewName != operation.Name)
```

Comment:
I'm still thinking that there should be a method to check for value changes, but okay, let's leave it as is for the compatibility reason.